staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -190,8 +226,18 @@ func ExtractList(obj runtime.Object) ([]runtime.Object, error) {
 			default:
 				list[i] = nil
 			}
-		case runtime.Object:
-			list[i] = item
+		case raw.Type().Implements(objectType):
+			list[i] = raw.Interface().(runtime.Object)
+		case allocNew:
+			// shallow copy to avoid retaining a reference to the original list item
+			itemCopy := reflect.New(raw.Type())
```

Comment:
can raw be an interface type here? if so, does this work properly?

Reply:
https://go.dev/play/p/ccO3C4IJ8Un